runtime/drivers/clickhouse/clickhouse.go

```diff
@@ -491,14 +511,26 @@ func (c *Connection) Config() map[string]any {
 func (c *Connection) Close() error {
 	c.cancel()
 
-	errDB := c.db.Close()
+	var errReadDB error
+	if c.readDB != nil {
```

Comment:
It's not possible to have an open connection that doesn't have this set. Lots of code relies on the `readDB` and `writeDB` not being nil, so it's weird to see checks like that here.